packages/react/src/constants/device-sizes.constant.ts

```diff
@@ -36,3 +38,39 @@ export const sizes = {
     return 'large';
   },
 };
+
+interface Breakpoints {
+  small: number;
+  medium: number;
+}
+
+export const getSizesForBreakpoints = ({ small, medium }: Breakpoints) => {
+  const newSizes = {
+    ...sizes, // Note: this helps get the function from sizes
+    ...fastClone(sizes), // Note: this helps to get a deep clone of fields like small, medium etc
+  };
+
+  if (!small || !medium) {
+    return newSizes;
+  }
+
+  newSizes.small = {
+    max: small,
+    min: Math.floor(small / 2),
+    default: newSizes.small.min + 1, // TODO: handle negative values?
+  };
+
+  newSizes.medium = {
+    max: medium,
+    min: newSizes.small.max + 1,
+    default: newSizes.medium.min + 1,
+  };
+
+  newSizes.large = {
+    max: 2000, // TODO: decide upper limit
+    min: newSizes.medium.max + 1,
+    default: newSizes.large.min + 1,
+  };
```

Comment:
the way you calculate the `default` values is giving the old values. You want to read `newSizes.small.min`, but you want the new value that still has not been set. You are instead reading the values that exist from cloning `sizes`.

See this example:

<img width="402" alt="image" src="https://user-images.githubusercontent.com/1393142/190497882-cd42adca-ae99-4e71-9156-566ec9b866a3.png">

As you can see, `bar` uses the old `foo` value, not the new one being set in the same expression.

You're going to want to first set the `max` values, then the `min` values, and then set the `default` values afterwards to make sure to read them correctly. Or use the `small`/`medium` variables instead:

```ts
// this doesn't work
  newSizes.large = {
    max: 2000, // TODO: decide upper limit
    min: newSizes.medium.max + 1,
    default: newSizes.large.min + 1,
  };

// this works
  newSizes.large = {
    max: 2000, // TODO: decide upper limit
    min: newSizes.medium.max + 1,
  };
newSizes.large.default = newSizes.large.min + 1;

// this also works
  newSizes.large = {
    max: 2000, // TODO: decide upper limit
    min: newSizes.medium.max + 1,
    default: newSizes.medium.max + 1 + 1,
  };
```

I recommend the middle approach to avoid duplicating the calculations

Reply:
That's a great catch. Thanks.